File: 0071 Simplify Path/0071.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        stk = []

        for w in path.split('/'):
            if w == '..':
                if stk:
                    stk.pop()
            elif w not in ['', '.']:
                stk.append(w)

        return f"/{'/'.join(stk)}"

File: 0071 Simplify Path/0071.cpp
class Solution {
public:
    string simplifyPath(string path) {
        stack<string> stk;
        string w;
        
        for (char c: path + "/"){
            if (c == '/'){
                if (w == ".."){
                    if (!stk.empty()) stk.pop();
                } else if (w != "" && w != "."){
                    stk.push(w);
                }
                w.clear();
            } else {
                w += c;
            }
        }

        string ans = "";

        while (!stk.empty()){
            ans = "/" + stk.top() + ans;
            stk.pop();
        }

        return ans == "" ? "/" : ans;
    }
};
